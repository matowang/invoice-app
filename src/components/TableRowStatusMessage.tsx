import { Button, TableCell, TableRow } from "@mui/material";
import ErrorOutlineOutlinedIcon from '@mui/icons-material/ErrorOutlineOutlined';
import HelpOutlineOutlinedIcon from '@mui/icons-material/HelpOutlineOutlined';
import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
import { ReactNode } from "react";

interface TableRowErrorProps {
    colSpan?: number;
    children?: ReactNode,
    status?: TableStatus;
}

type TableStatus = ('error' | 'empty' | 'default');

const TableStatusColor: Record<TableStatus, string> = {
    error: 'text-red-500',
    empty: 'text-neutral-500',
    default: 'text-neutral-700',
}

const TableStatusIcon: Record<TableStatus, ReactNode> = {
    error: <ErrorOutlineOutlinedIcon />,
    empty: <HelpOutlineOutlinedIcon />,
    default: <InfoOutlinedIcon />
}

const TableRowStatusMessage = ({ colSpan, children, status = 'default' }: TableRowErrorProps) => (
    <TableRow>
        <TableCell align="center" colSpan={colSpan} className={`${TableStatusColor[status]} py-32`}>
            <div className="flex items-center gap-2 justify-center">
                {TableStatusIcon[status]}<div>{children ?? "Something went wrong."}</div>
            </div>
        </TableCell>
    </TableRow>
);
export default TableRowStatusMessage;