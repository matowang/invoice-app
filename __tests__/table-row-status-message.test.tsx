import { render } from "@testing-library/react";
import TableRowStatusMessage from "../src/components/TableRowStatusMessage";

describe("table row message", () => {
	it("renders normal properly", () => {
		const { container } = render(<TableRowStatusMessage colSpan={5}>Status</TableRowStatusMessage>);
		expect(container).toMatchSnapshot();
	});
	it("renders empty properly", () => {
		const { container } = render(
			<TableRowStatusMessage colSpan={10} status='empty'>
				Status
			</TableRowStatusMessage>
		);
		expect(container).toMatchSnapshot();
	});
	it("renders error properly", () => {
		const { container } = render(
			<TableRowStatusMessage colSpan={3} status='error'>
				Status
			</TableRowStatusMessage>
		);
		expect(container).toMatchSnapshot();
	});
});
