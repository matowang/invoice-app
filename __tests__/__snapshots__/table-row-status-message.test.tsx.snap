// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table row message renders empty properly 1`] = `
<div>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-neutral-500 py-32 css-1yhpg23-MuiTableCell-root"
          colspan="10"
        >
          <div
            class="flex items-center gap-2 justify-center"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HelpOutlineOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
              />
            </svg>
            <div>
              Status
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`table row message renders error properly 1`] = `
<div>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-red-500 py-32 css-1yhpg23-MuiTableCell-root"
          colspan="3"
        >
          <div
            class="flex items-center gap-2 justify-center"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ErrorOutlineOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <div>
              Status
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
